### PR TITLE
Renamed documentation pages

### LINKED ISSUE #311: Align documentation headers and ToC entries
In [the documentation](https://junit-pioneer.org/docs/), some extensions use their annotations as titles and ToC entries (e.g. "@DisableIfDisplayName"), others use natural language in different capitalization (e.g. "Report entries" and "Range Sources"). This should be aligned when possible (and then documented). But what naming strategy works best?

* annotations or natural language?
* should ToC entries and page titles be identical?
* important to keep in mind that page title and description play a role for search engines

### PR DESCRIPTION
**Pull request by Michael1993**
Closes #311 

Signed-off-by: Mihaly Verhas <misi.verhas@gmail.com>
---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Comment by nicolaiparlog** [comment]
You didn't think this was going to be easy, did you? :grin: 

* on the issue, [I commented on capitalization](https://github.com/junit-pioneer/junit-pioneer/issues/311#issuecomment-669098138)
* the entries in `docs-nav.yml` need to be updated as well
* the `:page-description:` tags need to be updated to mention the actual annotations

On the latter, I recommend the following format (maximum 160 chars):

> Extends JUnit Jupiter with $ANNOTATIONS, which $DESCRIPTION.

E.g.:

> Extends JUnit Jupiter with `@IntRangeSource`, `@FloatRangeSource`, etc., which provide series of numbers as argument sources for a parameterized tests.

File: docs/docs-nav.yml [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
         url: /docs/default-locale-timezone/
-      - title: "@DisableIfDisplayName"
+      - title: "Disable Based on DisplayName"
         url: /docs/disable-if-display-name/
       - title: "Range Sources"
         url: /docs/range-sources/
-      - title: "@RetryingTest"
+      - title: "Retrying Failing Tests"
         url: /docs/retrying-test/
       - title: "Report entries"
```
Comment (nicolaiparlog) [inline docs/docs-nav.yml:17]:
```suggestion
      - title: "Publishing Report Entries"
```

Then the ToC needs to be resorted.

File: docs/report-entries.adoc [hunk 2]
```diff
@@ -1,5 +1,5 @@
 :page-title: Report entries
-:page-description: JUnit Jupiter extension to report with annotations.
+:page-description: Extends JUnit Jupiter with `@ReportEntry`, which reports entries using `TestReporter`.
```
Comment (nicolaiparlog) [inline docs/report-entries.adoc:2]:
```suggestion
:page-description: Extends JUnit Jupiter with `@ReportEntry`, which publishes reports entries using `TestReporter`.
```

File: docs/report-entries.adoc [hunk 3]
```diff
@@ -1,5 +1,5 @@
 :page-title: Report entries
```
Comment (nicolaiparlog) [inline docs/report-entries.adoc:1]:
```suggestion
:page-title: Publishing Report Entries
```

File: docs/temp-directory.adoc [hunk 4]
```diff
@@ -1,5 +1,5 @@
 :page-title: TempDirectory
```
Comment (nicolaiparlog) [inline docs/temp-directory.adoc:1]:
```suggestion
:page-title: Temporary Files and Directories
```

Same in the ToC.

**Comment by nicolaiparlog** [comment]
Thanks @Michael1993 for taking this on so quickly. As I see it, this can be merged.

**Comment by Bukama** [comment]
Damn, I forget the issue and PR numbers in the commit message body :(

**Pull request merged**